pkg/products/observability/reconciler.go

```diff
@@ -0,0 +1,369 @@
+		return integreatlyv1alpha1.PhaseInProgress, nil
+	}
+	return integreatlyv1alpha1.PhaseCompleted, nil
+
+}
+
+func (r *Reconciler) reconcileComponents(ctx context.Context, serverClient k8sclient.Client, productNamespace string) (integreatlyv1alpha1.StatusPhase, error) {
+
+	oo := &observability.Observability{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "observability-stack",
+			Namespace: productNamespace,
+		},
+	}
+
+	op, err := controllerutil.CreateOrUpdate(ctx, serverClient, oo, func() error {
+		disabled := true
+		oo.Spec = observability.ObservabilitySpec{
+			ConfigurationSelector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"monitoring-key": r.Config.GetLabelSelector(),
+				},
+				MatchExpressions: nil,
+			},
+			SelfContained: &observability.SelfContained{
+				DisableRepoSync:         &disabled,
+				DisableObservatorium:    &disabled,
+				DisablePagerDuty:        &disabled,
+				DisableDeadmansSnitch:   &disabled,
+				DisableBlackboxExporter: nil,
+				SelfSignedCerts:         nil,
+				FederatedMetrics:        nil,
+				PodMonitorLabelSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"monitoring-key": r.Config.GetLabelSelector(),
+					},
+				},
+				PodMonitorNamespaceSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"monitoring-key": r.Config.GetLabelSelector(),
+					},
+				},
+				ServiceMonitorLabelSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"monitoring-key": r.Config.GetLabelSelector(),
+					},
+				},
+				ServiceMonitorNamespaceSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"monitoring-key": r.Config.GetLabelSelector(),
+					},
+				},
+				RuleLabelSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"monitoring-key": r.Config.GetLabelSelector(),
+					},
+				},
+				RuleNamespaceSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"monitoring-key": r.Config.GetLabelSelector(),
+					},
+				},
+				ProbeLabelSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"monitoring-key": r.Config.GetLabelSelector(),
+					},
+				},
+				ProbeNamespaceSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"monitoring-key": r.Config.GetLabelSelector(),
+					},
+				},
+				AlertManagerConfigSecret: alertManagerConfigSecretName,
+			},
+			ResyncPeriod: "1h",
+		}
+
+		return nil
+	})
+	if err != nil {
+		return integreatlyv1alpha1.PhaseInProgress, err
+	}
+	if op == controllerutil.OperationResultUpdated || op == controllerutil.OperationResultCreated {
+		return integreatlyv1alpha1.PhaseInProgress, nil
+		//	r.log.Info(string(integreatlyv1alpha1.PhaseCompleted))
+		//} else {
+		//	return integreatlyv1alpha1.PhaseInProgress, nil
+	}
+
+	err = serverClient.Get(ctx, k8sclient.ObjectKey{Name: "observability-stack", Namespace: productNamespace}, oo)
```

Comment:
`observability-stack` is used in a few places, would it be better to put it as a constant ? :thinking:

Reply:
Fixed